### DISCUSSION TITLE
Why do enums computed properties show up in `metadata.descriptor.fields`?

### DISCUSSION THREAD

**Discussion by NSExceptional** [Azoy/Echo#33] [Q&A] [answered]
Enums cannot have stored properties, but computed properties show up in `.descriptor.fields`. My testing shows that computed properties do NOT show up in this variable for struct or class metadata. Why do they show up for enums? And where do they show up for structs and classes?

**Comment by Azoy** [comment, marked as answer]
Can you give me an example of computed properties showing up in the fields descriptor for enums? AFAIK computed properties should not be emitted there for structs, classes, or enums.

**Reply by NSExceptional** [reply]
Oh, the _cases_ show up as fields! I think in my case I had cases like `X`/`Y`/`Z` and computed properties like `toX`/`toY`/`toZ` and so I got confused and assumed it was listing the computed properties.

My bad! Disregard
